overlord/snapstate/snapstate_update_test.go

```diff
@@ -7086,7 +7162,127 @@ func (s *snapmgrTestSuite) TestUpdateWithHiddenSnapDataDir(c *C) {
 	c.Assert(chg.Err(), IsNil)
 	c.Assert(chg.Status(), Equals, state.DoneStatus)
 
-	c.Assert(s.fakeBackend.ops.MustFindOp(c, "copy-data").dirOpts, DeepEquals, &dirs.SnapDirOptions{HiddenSnapDataDir: true})
+	// shouldn't do migration since it was already done
+	c.Assert(s.fakeBackend.ops.First("hide-snap-data"), IsNil)
+	c.Assert(s.fakeBackend.ops.First("undo-hide-snap-data"), IsNil)
+
+	opts := &dirs.SnapDirOptions{HiddenSnapDataDir: true}
+	c.Assert(s.fakeBackend.ops.MustFindOp(c, "copy-data").dirOpts, DeepEquals, opts)
+
+	// check state and seq file still have correct migration state
+	assertMigrationState(c, s.state, "some-snap", true)
+}
+
+func (s *snapmgrTestSuite) TestUndoMigrationAfterHidingFails(c *C) {
+	s.testUndoMigration(c, false)
```

Comment:
it's hard to tell what a bare boolean argument means, so let's try to avoid that:
```suggestion
	const failInUndo = false
	s.testUndoMigration(c, failInUndo)
```

Reply:
Fixed both in latest commit. Thanks